File: Enkle algoritmer/twoSum/twoSum.py
import math
import random

nums = [4,2,7,5,4,12,18]
target = int(9)

#Finner de som er etterhverandre.
def twoSumBasic(a, b):
    output = []
    for i in range(0, len(a), 1):
        j = i+1
        if j<7:
            if(a[i] + a[j] == b):
                output.append([[i,j], [a[i], a[j]]])
    return output
# print(twoSumBasic(nums, target))

#Finner de uansett rekkefølge
def rndNum():
    tall = random.randint(0,20)
    return tall

tallListe = [rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum()]

print("tallListe:")
print(tallListe)
print("----------------------------------------------")
print("Skriv inn target, mellom 1 og 20")
targetNum = int(input("target: "))

print("----------------------------------------------")
print("twosum resutat:")
def twoSumAdvanced(a, b):
    output = []
    for i in range(0, len(a), 1):
        for j in range(0, len(a), 1):
            if a[i] + a[j] == b:
                output.append([[i,j], [a[i], a[j]]])
    return output
print(twoSumAdvanced(tallListe, targetNum))

File: Enkle algoritmer/twoSum/twoSum.js
//Finner de som er etterhverandre.
let tall = [2,5,5,11];
let tgt = 10;

var twoSum = function(nums, target) {
    let output = [];
    let funnet = false;
    while(funnet === false){
        for(let i=0; i<nums.length; i++){
            for(let j=1; j<nums.length; j++){
                if(nums[i] + nums[j] === target){
                    if(i != j){
                        output.push(i, j);
                        output.length = 2;
                        funnet = true;
                    }
                };
            }
        }
    }
    return output
};

console.log(twoSum(tall, tgt));

// console.log("-----Sum etterhverandre i Arry----");
// console.log(twoSum(nums, tgt));
// console.log("----------------------------------");

// // // let tallListe = [1,4,7,5,2];
// const rndNum = () => Math.floor(Math.random() * 20) + 1;
// let tallListe = [rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum(), rndNum()];
// let target = 12;
// console.log(tallListe)

function finSum(a, b){
    let output = [];
    for(let i=0; i<a.length; i++){
        for(let j=0; j<a.length; j++){
            if(a[i] + a[j] === b){output.push([[i, j], [a[i], a[j]]])}
        }
    }
    return output
}

// console.log(finSum(tallListe, target));
// console.log([rndNum(), rndNum()])